src/content/en/2019/javascript.md

```diff
@@ -0,0 +1,338 @@
+
+At the median, **89%** more third-party code is used than first-party code authored by the developer
+for both mobile and desktop**. **This clearly shows that third-party code can be one of the biggest
+contributors to bloat.
+
+# Resource compression
+
+In the context of browser-server interactions, **resource compression** refers to code that has been
+modified using a data compression algorithm. Resources can be compressed statically ahead of time or
+on-the-fly as they are requested by the browser, and for either approach the transferred resource
+size is significantly reduced which improves page performance.
+
+There are multiple text-compression algorithms, but only two are mostly used for the compression
+(and decompression) of HTTP network requests:
+
+-  [Gzip](https://www.gzip.org/) (gzip): The most widely used compression format for server and
+   client interactions
+-  [Brotli](https://github.com/google/brotli) (br): A newer compression algorithm aiming to
+   further improve compression ratios. [90% of browsers](https://caniuse.com/#feat=brotli) support
+   Brotli encoding.
+
+Compressed scripts will always need to be uncompressed by the browser once transferred. This means
+its content remains the same and execution times are not optimized whatsoever. Resource compression,
+however, will always improve download times which also is one of the most expensive stages of
+JavaScript processing. Ensuring JavaScript files are compressed correctly can be one of the most
+significant factors in improving site performance. 
+
+**How many sites are compressing their JavaScript resources**?
+
+<insert graphics of metric 01_05 and 01_06>
+
+The majority of sites are compressing their JavaScript resources. Gzip encoding is used on
+**~64-67%** of sites and Brotli on **~14%**. Compression ratios are similar for both desktop and
+mobile.
+
+For a deeper analysis on compression, refer to the ["Compression"](/compression) chapter.
+
+# Open source libraries and frameworks
+
+Open source code, or code with a permissive license that can be accessed, viewed and modified by
+anyone. From tiny libraries to entire browsers, such as [Chromium](https://www.chromium.org/Home)
+and [Firefox](https://www.openhub.net/p/firefox), open source code plays a crucial role in the world
+of web development. In the context of JavaScript, developers rely on open source tooling to include
+all types of functionality into their web page. Regardless of whether a developer decides to use a
+small utility library or a massive framework that dictates the architecture of their entire
+application, relying on open-source packages cam make feature development easier and faster.
+
+**Which JavaScript open-source libraries are used the most?**
+
+<insert graphics of metric 01_08>
+
+[jQuery](https://jquery.com/), the most widely used JavaScript library ever created, is used in
+**85.03%** of desktop pages and **83.46%** of mobile pages. The advent of many Browser APIs and
```

Comment:
Do we have data on jQuery versions or sizes? AFAIK the current / latest version is a much thinner and modular wrapper than the old one, and it could be interesting to see which one websites are ending up using.